src/main/java/uk/ac/ebi/ena/readtools/loader/common/eater/AbstractPagedDataConsumer.java

```diff
@@ -31,8 +31,8 @@
 
 import uk.ac.ebi.ena.readtools.loader.common.Pair;
 
-public abstract class 
-AbstractPagedDataEater<T1, T2> extends AbstractDataEater<T1, T2>
+public abstract class
+AbstractPagedDataConsumer<T1, T2> extends AbstractDataConsumer<T1, T2>

```

Comment:
T1 and T2 should be restricted to concrete types (e.g. marker interfaces). We should log rather than write to System.out.